batch/batch/aioclient.py

```diff
@@ -0,0 +1,328 @@
+import math
+import time
+import random
+import yaml
+import cerberus
+import aiohttp
+
+from . import schemas
+
+
+class Job:
+    def __init__(self, client, id, attributes=None, parent_ids=None, scratch_folder=None, _status=None):
+        if parent_ids is None:
+            parent_ids = []
+        if attributes is None:
+            attributes = {}
+
+        self.client = client
+        self.id = id
+        self.attributes = attributes
+        self.parent_ids = parent_ids
+        self.scratch_folder = scratch_folder
+        self._status = _status
+
+    async def is_complete(self):
+        if self._status:
+            state = self._status['state']
+            if state in ('Complete', 'Cancelled'):
+                return True
+        await self.status()
+        state = self._status['state']
+        return state in ('Complete', 'Cancelled')
+
```

Comment:
Did you intentionally remove `cached_status`?

Reply:
Yes.